src/steps/zsh.rs

```diff
@@ -177,26 +177,23 @@ pub fn run_oh_my_zsh(ctx: &ExecutionContext) -> Result<()> {
     // the ZSH variable for topgrade here.
     if ctx.under_ssh() {
         let zshrc_path = zshrc().require()?;
-        // `res_output` will be an `Err` if `ZSH` is not present (propagated by the `grep` command)
-        let res_output = Command::new("zsh")
+        let output = Command::new("zsh")
             .args([
                 "-c",
                 // ` > /dev/null` is used in case the user's zshrc will have some stdout output.
-                format!(
-                    "source {} > /dev/null && export -p | grep -q ZSH && echo $ZSH",
-                    zshrc_path.display()
-                )
-                .as_str(),
+                format!("source {} > /dev/null && export -p", zshrc_path.display()).as_str(),
             ])
-            .output_checked_utf8();
-
-        if let Ok(output) = res_output {
-            if output.status.success() {
-                // trim the newline char
-                let zsh_env = output.stdout.trim();
-                if !zsh_env.is_empty() {
-                    env::set_var("ZSH", zsh_env);
-                }
+            .output_checked_utf8()?;
+
+        let stdout = output.stdout;
+
+        for line in stdout.lines() {
+            let prefix = "export ZSH=";
```

Comment:
since the prefix variable doesn't change, wouldn't it make more sense to declare it outside of the for loop?

Reply:
Good catch! Thanks!

Reply:
done